### PR TITLE
Fix for zero axes or zero buttons

### PR DESCRIPTION
**Pull request by tzarc**
Figured it'd be good to have something available to test this feature.

### CODE REVIEW

**Comment by tzarc** [comment]
`make handwired/onekey/promicro:joystick`
`make handwired/onekey/proton_c:joystick`

**Comment by tzarc** [comment]
Fixed up the device descriptor to allow for zero axes or zero buttons.
If either were zero the device wasn't loading correctly.

**Comment by tzarc** [comment]
With the USB descriptor change:

image

image

File: quantum/process_keycode/process_joystick.c [hunk 1]
```diff
@@ -9,6 +9,7 @@
 #endif
 
 #include <string.h>
+#include <math.h>
```
Comment (tzarc) [inline quantum/process_keycode/process_joystick.c:12]: This fixes ARM builds

**Comment by a-chol** [comment]
Nice catch, thanks for the fix!

**Pull request merged**